src/config/mod.rs

```diff
@@ -81,4 +81,18 @@ impl Config {
         let contents = tokio::fs::read_to_string(path).await?;
         Config::from_toml_str(contents.as_str())
     }
+
+    pub async fn write_to_file_async(&self, path: impl AsRef<Path>) -> Result<()> {
```

Comment:
Just a quick check: Why do we need this function?

Reply:
yep yep